disnake/threads.py

```diff
@@ -208,22 +208,24 @@ def _from_data(self, data: ThreadPayload) -> None:
             self.me = ThreadMember(self, member)
 
     def _unroll_metadata(self, data: ThreadMetadata) -> None:
-        self.archived = data["archived"]
-        self.auto_archive_duration = data["auto_archive_duration"]
-        self.archive_timestamp = parse_time(data["archive_timestamp"])
-        self.locked = data.get("locked", False)
-        self.invitable = data.get("invitable", True)
-        self.create_timestamp = parse_time(data.get("create_timestamp"))
-
-    def _update(self, data) -> None:
+        self.archived: bool = data["archived"]
+        self.auto_archive_duration: ThreadArchiveDurationLiteral = data["auto_archive_duration"]
+        self.archive_timestamp: datetime.datetime = parse_time(data["archive_timestamp"])
+        self.locked: bool = data.get("locked", False)
+        self.invitable: bool = data.get("invitable", True)
+        self.create_timestamp: Optional[datetime.datetime] = parse_time(
+            data.get("create_timestamp")
+        )
+
+    def _update(self, data: ThreadPayload) -> None:
         try:
-            self.name = data["name"]
+            self.name: str = data["name"]
         except KeyError:
             pass
 
-        self.slowmode_delay = data.get("rate_limit_per_user", 0)
-        self._flags = data.get("flags", 0)
-        self._applied_tags = list(map(int, data.get("applied_tags", [])))
+        self.slowmode_delay: int = data.get("rate_limit_per_user", 0)
+        self._flags: int = data.get("flags", 0)
+        self._applied_tags: List[int] = list(map(int, data.get("applied_tags", [])))
```

Comment:
These attributes already have annotations in `__init__`, there isn't really a need to redeclare the types here

Reply:
I see

Reply:
1ea5d74973f637f47ce4a6bb8f2a3c80af1f456d